test/controllers/v1/profiles_controller_test.rb

```diff
@@ -11,28 +11,65 @@ class ProfilesControllerTest < ActionDispatch::IntegrationTest
-      profiles(:one).update! rules: [rules(:one)]
-      get tailoring_file_v1_profile_url(profiles(:one).id)
-      assert_response :no_content
-    end
+    class TailoringFileTest < ProfilesControllerTest
+      test 'tailoring_file with a canonical profile returns no content' do
+        profiles(:one).update! rules: [rules(:one)]
+        get tailoring_file_v1_profile_url(profiles(:one).id)
+        assert_response :no_content
+      end
+
+      test 'tailoring_file with a noncanonical profile matching its parent'\
+        'returns no content' do
+        profiles(:two).update!(rules: [rules(:one)])
+        profiles(:one).update!(parent_profile: profiles(:two),
+                               rules: [rules(:one)])
+        get tailoring_file_v1_profile_url(profiles(:one).id)
+        assert_response :no_content
+      end
 
-    test 'tailoring_file with a noncanonical profile matching its parent'\
-      'returns no content' do
-      profiles(:two).update!(rules: [rules(:one)])
-      profiles(:one).update!(parent_profile: profiles(:two),
-                             rules: [rules(:one)])
-      get tailoring_file_v1_profile_url(profiles(:one).id)
-      assert_response :no_content
+      test 'tailoring_file with a noncanonical profile '\
+           'returns tailoring file' do
+        profiles(:two).update!(rules: [rules(:one), rules(:two)])
+        profiles(:one).update!(parent_profile: profiles(:two),
+                               rules: [rules(:one)])
+        get tailoring_file_v1_profile_url(profiles(:one).id)
+        assert_response :success
+        assert_equal Mime[:xml].to_s, @response.content_type
+      end
     end
 
-    test 'tailoring_file with a noncanonical profile returns tailoring file' do
-      profiles(:two).update!(rules: [rules(:one), rules(:two)])
-      profiles(:one).update!(parent_profile: profiles(:two),
-                             rules: [rules(:one)])
-      get tailoring_file_v1_profile_url(profiles(:one).id)
-      assert_response :success
-      assert_equal Mime[:xml].to_s, @response.content_type
+    class DestroyTest < ProfilesControllerTest
+      require 'sidekiq/testing'
+
+      test 'destroy an existing, accessible profile' do
+        profile_id = profiles(:one).id
+        delete profile_path(profile_id)
+        assert_response :success
+        assert_equal 202, response.status, 'Response should be 202 accepted'
+        assert_equal profile_id, JSON.parse(response.body).dig('data', 'id'),
+                     'Profile ID did not match deleted profile'
+      end
+
+      test 'v1 destroy an existing, accessible profile' do
+        profile_id = profiles(:one).id
+        delete v1_profile_path(profile_id)
+        assert_response :success
+        assert_equal 202, response.status, 'Response should be 202 accepted'
+        assert_equal profile_id, JSON.parse(response.body).dig('data', 'id'),
+                     'Profile ID did not match deleted profile'
+      end
```

Comment:
These should be wrapped in an `assert_difference` or something to verify the profile was truly deleted after the action. You may also want to check `DeleteHost.jobs.size` and call `DeleteHost.drain` to ensure the orphaned associated hosts are removed with the Profile - your call (we already check this in the ProfileHost tests), but if you don't check any jobs, I don't think sidekiq/testing is required

Reply:
I'll add those checks. Anytime a test queues any job in Sidekiq, `sidekiq/testing` must be required or the server tries to connect to redis and fails.

Reply:
So, part of the reason I didn't add this is because the controller has no knowledge of how the host gets deleted - it just calls `destroy` and the logic passes into the model, where there are unit tests to test host deletion. Whether a job is queued or whatever happens shouldn't be tested by the controller IMO. I think it's okay to test that the profile was deleted, though, just not how it was deleted.

Reply:
Agreed - I think I was not getting the error because I did have redis on while testing it